api/controllers/zetter.ts

```diff
@@ -1,294 +1,328 @@
+            user: true,
+            favorities: true,
+        },
+        orderBy: {
+            createdAt: 'desc',
+        },
+        take: limit,
+    });
+    res.json(followingUsersTweetsList);
 };
 
-export const getSpecificUsersTweets = (req: Request, res: Response) => {
+export const getSpecificUsersTweets = async (req: Request, res: Response) => {
     const limit: number = 10;
-    const user: ProfileType | undefined = users.find(({ username }) => username === req.params.username);
-    const sortedTweets: Array<TweetType> = tweets.sort(function (a: TweetType, b: TweetType) {
-        if (Number(a.createdAt) - Number(b.createdAt)) {
-            return -1;
-        } else {
-            return 1;
-        }
+    const user: User | null = await prisma.user.findUnique({
+        where: { username: req.params.username },
     });
 
     if (user) {
-        const specificUsersTweets: Array<TweetType> = sortedTweets.filter((tweet) => tweet.createdBy === user.id);
-        const result: Array<TweetType> = addInformationToTweet(specificUsersTweets, req);
-        res.json(result.slice(0, limit));
+        const specificUsersTweets = await prisma.tweet.findMany({
+            where: {
+                createdBy: user.id,
+            },
+            include: {
+                user: true,
+                favorities: true,
+            },
+            orderBy: {
+                createdAt: 'desc',
+            },
+            take: limit,
+        });
+
+        res.json(specificUsersTweets);
     }
 };
 
-export const getSpecificUsersFavoriteTweets = (req: Request, res: Response) => {
+export const getSpecificUsersFavoriteTweets = async (req: Request, res: Response) => {
     const limit: number = 10;
-    const user: ProfileType | undefined = users.find(({ username }) => username === req.params.username);
+    const user: User | null = await prisma.user.findUnique({
+        where: { username: req.params.username },
+    });
     if (!user) {
         throw Error;
     }
-    const favoriteTweets: Array<FavoriteType> = favorities.filter((favorite) => favorite.userId === user.id);
-    const favoriteTweetIds: Array<number> = favoriteTweets.map((obj) => obj.tweetId);
 
-    let favoriteTweetList: Array<TweetType> = [];
-
-    tweets.forEach((tweet) => {
-        if (favoriteTweetIds.includes(tweet.id)) {
-            favoriteTweetList.push(tweet);
-        }
+    const favoriteTweets = await prisma.favorite.findMany({
+        where: { userId: user.id },
     });
+    const favoriteTweetIds: Array<number> = favoriteTweets.map((obj) => obj.tweetId);
 
-    const sortedFavoriteTweetList: Array<TweetType> = favoriteTweetList.sort(function (a, b) {
-        if (a.createdAt > b.createdAt) {
-            return -1;
-        } else {
-            return 1;
-        }
+    const favoriteTweetList: Array<Tweet> = await prisma.tweet.findMany({
+        where: {
+            id: {
+                in: favoriteTweetIds,
+            },
+        },
```

Comment:
favoriteIdsを一回とってこなくても、これで一発でとれるよ

https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#relation-filters

```suggestion
        where: {
            favorities: {
                some: { userId: user.id }
            },
        },
```